### PR TITLE
[SYCL] Sync SYCL with latest CUDA backend

### PR DESCRIPTION
**Pull request by abagusetty**
Tests verified on Nvidia, Intel GPUs

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -30,6 +30,12 @@ if( EXCHCXX_ENABLE_SYCL AND EXCHCXX_ENABLE_HIP )
 endif()
 
 
+if(EXCHCXX_ENABLE_SYCL)
+  # e.g. intel_gpu_pvc | nvidia_gpu_sm_80 | nvidia_gpu_sm_90 | amd_gpu_gfx90a | amd_gpu_gfx942
+  set(EXCHCXX_SYCL_TARGET "" CACHE STRING "Alias for -fsycl-targets (see Users Manual)")
```
Comment (wavefunction91) [inline CMakeLists.txt:35]: Shouldn't this be an `option` defaulted to `""`?

File: include/exchcxx/impl/builtin/kernels/deorbitalized.hpp [hunk 2]
```diff
@@ -141,8 +141,10 @@ struct kernel_traits<Deorbitalized<XCEF,KEDF>> {
                          double& v2rho2, double& v2rhosigma, double& v2rholapl, double& v2rhotau,
                          double& v2sigma2, double& v2sigmalapl, double& v2sigmatau,
                          double& v2lapl2, double& v2lapltau, double& v2tau2 ) {
-    #if defined(__CUDACC__) || defined(__HIPCC__) ||  defined(__SYCL_DEVICE_ONLY__) || defined(EXCHCXX_ENABLE_SYCL)
+    #if defined(__CUDACC__) || defined(__HIPCC__)
     printf("eval_vxc_fxc_unpolar not implemented for deorbitalized kernels\n");
+    #elif defined(__SYCL_DEVICE_ONLY__) || defined(EXCHCXX_ENABLE_SYCL)
+    sycl::ext::oneapi::experimental::printf("eval_vxc_fxc_unpolar not implemented for deorbitalized kernels\n");
```
Comment (wavefunction91) [inline include/exchcxx/impl/builtin/kernels/deorbitalized.hpp:147]: This brings up an interesting point - this is very `openapi` specific. Is it safe to say that we're not supporting SYCL in general, but rather `oneapi`? Is there a way to make this SYCL-only compliant?

File: include/exchcxx/impl/builtin/util.hpp [hunk 3]
```diff
@@ -139,11 +139,16 @@ SAFE_INLINE(auto) xc_cheb_eval(const double x, const double *cs, const int N)
 
   return 0.5*(b0 - b2);
 }
+
 // The following data is taken from libxc
 #if defined(__CUDACC__) || defined(__HIPCC__)
-__device__
+__device__ static
+#elif defined(__SYCL_DEVICE_ONLY__)
```
Comment (wavefunction91) [inline include/exchcxx/impl/builtin/util.hpp:146]: This should be a separate (reused) define to reduce code bloat.

File: CMakeLists.txt [hunk 4]
```diff
@@ -69,7 +75,7 @@ if( ${Libxc_FOUND} )
 else()
 
   option( FETCHCONTENT_LIBXC_GIT_SHALLOW      "Whether to use GIT_SHALLOW for FetchContent'ing libxc"     ON )
-  
+
```
Comment (wavefunction91) [inline CMakeLists.txt:78]: Can you go through and carefully remove the white-space-only changes? I'll be adding a linter to this code base at some point, but I don't see a reason to change things before then.

**Review by wavefunction91 — CHANGES_REQUESTED** [review]
I still need to look at the tests, but that will be easier once you address the whitespace comment.

File: include/exchcxx/impl/builtin/kernels/deorbitalized.hpp (hunk 2 above)
Comment (abagusetty) [reply include/exchcxx/impl/builtin/kernels/deorbitalized.hpp:147]: For the printf currently to work in the SYCL device-side, unfortunately we have to use `sycl::ext::oneapi::experimental::printf`. Doesnt work with AdaptiveCPP. Updated README accordingly. There is a generic way using `sycl::stream` but definitely not a cleaner solution because of command-grp object

File: CMakeLists.txt (hunk 1 above)
Comment (abagusetty) [reply CMakeLists.txt]: I've changed the archs to `option`

File: CMakeLists.txt [hunk 5]
```diff
@@ -12,6 +12,13 @@ option( EXCHCXX_ENABLE_SYCL      "Enable Device Code (SYCL)"    OFF )
 option( EXCHCXX_ENABLE_LIBXC     "Enable Libxc Backend"         ON  )
 option( BUILD_SHARED_LIBS        "Build Shared Libs"            OFF )
 
+# --- SYCL Architecture Options (mutually exclusive) ---
```
Comment (wavefunction91) [inline CMakeLists.txt:15]: Why is this listed out explicitly? Shouldn't this just be a variable that's compared to an available list? i.e.
```
if(EXCHCXX_SYCL_ARCH)
  ... # check if it's allowed
endif()
```

Seems that the current approach doesn't scale

File: include/exchcxx/impl/builtin/kernels/deorbitalized.hpp [hunk 6]
```diff
@@ -1,30 +1,30 @@
 /**
- * ExchCXX
+ * ExchCXX
```
[the remark this answers is not in the document]
Comment (abagusetty) [reply include/exchcxx/impl/builtin/kernels/deorbitalized.hpp]: Done!

File: CMakeLists.txt (hunk 5 above)
Comment (abagusetty) [reply CMakeLists.txt]: Sure, reverted it back to setting the parameter for AoT from cmd-line rather than an option(ON/OFF)

**Review by wavefunction91 — APPROVED** [review]
